library/include/flow/owning_process_id.hpp

```diff
@@ -11,25 +13,41 @@ namespace flow {
 struct owning_process_id
 {
     static constexpr auto default_process_id = invalid_process_id;
+    static constexpr auto default_status = wait_unknown_status{};
 
-    static auto fork() -> owning_process_id;
+    static auto fork() -> reference_process_id;
 
-    owning_process_id() noexcept = default;
+    owning_process_id();
+    owning_process_id(reference_process_id id);
     owning_process_id(const owning_process_id& other) = delete;
     owning_process_id(owning_process_id&& other) noexcept;
     ~owning_process_id();
```

Comment:
clang-tidy seems to be unaware of needing the destructor defaulted in .cpp to avoid `std::unique_ptr` running into an "incomplete" type with `owning_process_id::impl`.